csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs

```diff
@@ -284,206 +288,247 @@ private void TestModelInputFloat()
         {
             // model takes 1x5 input of fixed type, echoes back
             string modelPath = Path.Combine(Directory.GetCurrentDirectory(), "test_types_FLOAT.pb");
-            var session = new InferenceSession(modelPath);
-            var container = new List<NamedOnnxValue>();
-            var tensorIn = new DenseTensor<float>(new float[] { 1.0f, 2.0f, -3.0f, float.MinValue, float.MaxValue }, new int[] { 1, 5 });
-            var nov = NamedOnnxValue.CreateFromTensor("input", tensorIn);
-            container.Add(nov);
-            var res = session.Run(container);
-            var tensorOut = res.First().AsTensor<float>();
-            Assert.True(tensorOut.SequenceEqual(tensorIn));
-            session.Dispose();
+
+            using (var session = new InferenceSession(modelPath))
+            {
+                var container = new List<NamedOnnxValue>();
+                var tensorIn = new DenseTensor<float>(new float[] { 1.0f, 2.0f, -3.0f, float.MinValue, float.MaxValue }, new int[] { 1, 5 });
+                var nov = NamedOnnxValue.CreateFromTensor("input", tensorIn);
+                container.Add(nov);
+                using (var res = session.Run(container))
+                {
+                    var tensorOut = res.First().AsTensor<float>();
+                    Assert.True(tensorOut.SequenceEqual(tensorIn));
+                }
+            }
         }
 
         [Fact(Skip = "Boolean tensor not supported yet")]
         private void TestModelInputBOOL()
         {
             // model takes 1x5 input of fixed type, echoes back
             string modelPath = Path.Combine(Directory.GetCurrentDirectory(), "test_types_BOOL.pb");
-            var session = new InferenceSession(modelPath);
-            var container = new List<NamedOnnxValue>();
-            var tensorIn = new DenseTensor<bool>(new bool[] { true, false, true, false, true }, new int[] { 1, 5 });
-            var nov = NamedOnnxValue.CreateFromTensor("input", tensorIn);
-            container.Add(nov);
-            var res = session.Run(container);
-            var tensorOut = res.First().AsTensor<bool>();
-            Assert.True(tensorOut.SequenceEqual(tensorIn));
-            session.Dispose();
+            using (var session = new InferenceSession(modelPath))
+            {
+                var container = new List<NamedOnnxValue>();
+                var tensorIn = new DenseTensor<bool>(new bool[] { true, false, true, false, true }, new int[] { 1, 5 });
+                var nov = NamedOnnxValue.CreateFromTensor("input", tensorIn);
+                container.Add(nov);
+                using (var res = session.Run(container))
+                {
+                    var tensorOut = res.First().AsTensor<bool>();
+                    Assert.True(tensorOut.SequenceEqual(tensorIn));
+                }
+            }
```

Comment:
Should we try to dispose a session as well?

Reply:
session is disposed by the using (var session = ){} block.